### DISCUSSION TITLE
Support for Sous Vide

### DISCUSSION THREAD

**Discussion by Toby-Mills** [andrei-tatar/node-red-contrib-smartnora#130] [Ideas]
I see that Google supports Sous Vide as a device type. Any chance of adding support in Smart Nora? Currently I have my homemade sous vide registered as a thermostat, which doesn't work very well.

**Comment by andrei-tatar** [comment]
Which of the traits would you use? Only on/off?
https://developers.google.com/assistant/smarthome/guides/sousvide

**Reply by Toby-Mills** [reply]
- On/Off
- Temperature Control
- Start/Stop
- Timer
would all be useful to me

**Reply by Toby-Mills** [reply]
Also, I'd be happy to help with coding it, but I'm not sure how I could do any testing?

**Reply by andrei-tatar** [reply]
@Toby-Mills In principle, all traits and handlers are defined in [NORA Common package](https://github.com/andrei-tatar/nora-firebase-common). Once they are defined in there, I integrate the package in the backend and it can be tested.

On/Off and Temperature Control traits are already supported so the change is only to add a new device type. The other traits are not yet supported.

**Reply by Toby-Mills** [reply]
looking at the code, it seems like:
in **nora-firebase-common**
add SousVide (in type DeviceType)
add a new trait (e.g. export type TimerDevice)

and then in **node-red-contrib-smartnora**
add nora-sousvide.html (for the config UI)
add nora-sousvide.ts (for the implementation logic)

does that sound right?

**Reply by andrei-tatar** [reply]
Pretty much. In `nora-firebase-common` there's also the command handlers (`src/execute/handlers.ts` file). And once it's done and I review it and merge it, it's possible to test it directly from node-red.

**Reply by andrei-tatar** [reply]
@Toby-Mills nora-firebase-common version 1.7.0 was published and integrated in the backend. Thanks for the contribution!
